src/pycel/lib/text.py

```diff
@@ -489,9 +489,46 @@ def mid(text, start_num, num_chars):
     #   mid-midb-functions-d5f9e25c-d7d6-472e-b568-4ecb12433028
 
 
-# def numbervalue(text):
+@excel_helper(cse_params=-1, str_params=(0, 1, 2))
+def numbervalue(text, decimal_separator=".", group_separator=","):
     # Excel reference: https://support.microsoft.com/en-us/office/
     #   numbervalue-function-1b05c8cf-2bfa-4437-af70-596c7ea7d879
+    if not text:
+        return 0
+    clean_text = text.replace(" ", "")
+
+    # TODO:
+    # If the decimal_separator and group_separator arguments are not specified,
+    # separators from the current locale are used.
+    if(decimal_separator and
+       group_separator and
+       clean_text.rfind(group_separator) > clean_text.rfind(decimal_separator)):
```

Comment:
If group and decimal separators are the same, Excel returns `#VALUE!`.
```suggestion
       clean_text.rfind(group_separator) >= clean_text.rfind(decimal_separator)):
```